src/main/java/seedu/address/model/module/Module.java

```diff
@@ -0,0 +1,95 @@
+package seedu.address.model.module;
+
+import java.util.Objects;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.group.Group;
+import seedu.address.model.group.UniqueGroupList;
+
+/**
+ * Represents a module in the TAT.
+ */
+public class Module {
+    private final String identifier;
+    private final UniqueGroupList groups;
+
+    /**
+     * Constructs a group object.
+     *
+     * @param identifier identifies the module. Usually equal
+     *                   to the module code.
+     */
+    public Module(String identifier) {
+        this.identifier = identifier;
+        groups = new UniqueGroupList();
+    }
+
+    //TODO: the following methods
+    /**
+     * Adds a group to groups.
+     */
+    public void addGroup(Group group) {
+        groups.add(group);
+    }
+
+    /**
+     * Gets group with given identifier.
+     */
```

Comment:
The comment is a bit unclear. At first thought that the identifier should be an integer.

Perhaps it could be: "Returns the group with the given id that is assigned to this module"
(this could be phrased better).

Also it should mention that null is returned if none of the groups have the given id.